src/screens/TopicPageScreen/elements/NavHeader.js

```diff
@@ -141,6 +166,25 @@ const NavHeader = (props) => {
         style={[styles.domain(isHeaderHide), {bottom: getBottomPostition()}]}>
         <TopicDomainHeader {...props} />
       </View>
+
+      {hasSearch && (
+        <View onLayout={onSearchLayout} style={styles.search}>
+          <Search
+            getSearchLayout={setSearchHeight}
+            searchText={searchText}
+            setSearchText={setSearchText}
+            setDiscoveryLoadingData={setDiscoveryLoadingData}
+            isFocus={isFocus}
+            setIsFocus={setIsFocus}
+            fetchDiscoveryData={fetchDiscoveryData}
+            onCancelToken={onCancelToken}
+            placeholderText={placeholderText}
+            setIsFirstTimeOpen={setIsFirstTimeOpen}
+            hideBackIcon={true}
+            autoFocus={false}
+          />
+        </View>
+      )}
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

A new conditional rendering block is added for the `Search` component. Ensure that all props passed to `Search` are correctly provided where this component is used.
```diff
+      {hasSearch && (
+        <View onLayout={onSearchLayout} style={styles.search}>
+          <Search
+            getSearchLayout={setSearchHeight}
+            searchText={searchText}
+            setSearchText={setSearchText}
+            setDiscoveryLoadingData={setDiscoveryLoadingData}
+            isFocus={isFocus}
+            setIsFocus={setIsFocus}
+            fetchDiscoveryData={fetchDiscoveryData}
+            onCancelToken={onCancelToken}
+            placeholderText={placeholderText}
+            setIsFirstTimeOpen={setIsFirstTimeOpen}
+            hideBackIcon={true}
+            autoFocus={false}
+          />
+        </View>
+      )}
```


<!-- This is an auto-generated comment by OSS CodeRabbit -->